src/physics.cpp

```diff
@@ -148,20 +69,150 @@ void Physics::step(Scene *scene, float dt)
+    for (int i = 0; i < constraints.planeCollisionConstraints.cardinality; i++) 
+    { 
+        numBoundConstraints[constraints.planeCollisionConstraints.particles[i]]--;
+    }
+    for (int i = 0; i < constraints.particleCollisionConstraints.cardinality; i++)
+    { 
+        numBoundConstraints[constraints.particleCollisionConstraints.particles[i][0]]--;
+        numBoundConstraints[constraints.particleCollisionConstraints.particles[i][1]]--;
+    }
+    constraints.planeCollisionConstraints.clear();
+    constraints.particleCollisionConstraints.clear();
 	
 }
+
+void Physics::resolveConstraints(std::vector<glm::vec3> & pPosition, std::vector<float> & invmass, std::vector<int> & numBoundConstraints)
+{
+    for (int i = 0; i < iterations; i++)
+    {
+
+        /**
+        * Distance Constraints
+        */
+        DistanceConstraintData &distanceConstraints = constraints.distanceConstraints;
+        vec3 delta1(0);
+        vec3 delta2(0);
+        for (int constraintIndex = 0; constraintIndex < distanceConstraints.cardinality; constraintIndex++)
+        {
+            if (distanceConstraints.solveDistanceConstraint(delta1, delta2, constraintIndex, particles))
+            {
+                // delta p_i = -w_i * s * grad_{p_i} C(p) * stiffness correction 
+                ivec2 &constraintParticles = distanceConstraints.particles[constraintIndex];
+                int p1 = constraintParticles[0];
+                int p2 = constraintParticles[1];
+                pPosition[p1] -= 
+                    delta1
+                    * (1 - pow(1 - distanceConstraints.stiffness[constraintIndex], 1 / (float)i))
+                    * overRelaxConst
+                    / (float)numBoundConstraints[p1];
+
+                pPosition[p2] -= 
+                    delta2 
+                    * (1 - pow(1 - distanceConstraints.stiffness[constraintIndex], 1 / (float)i))
+                    * overRelaxConst
+                    / (float)numBoundConstraints[p2];
+            }
+        }
+        
+        /**
+        *
+        */
+    }
+}
+
+void Physics::dampPlaneCollision(std::vector<int> & numBoundConstraints, std::vector<glm::vec3> & velocity, PlaneCollisionConstraintData & planeConstraints)
+{
+    for (int i = 0; i < planeConstraints.particles.size(); i++)
+    {
+        // Split into tangential and normal velocity
+        vec3 nVel = dot(planeConstraints.normals[i], velocity[planeConstraints.particles[i]]) * planeConstraints.normals[i];
+        vec3 tVel = -nVel + velocity[planeConstraints.particles[i]];
+        // Damp tangential velocity according to parameter
+        tVel *= restitutionCoefficientT;
+        // flip normal velocity
+        nVel = dot(nVel, planeConstraints.normals[i]) > 0 ? nVel : -nVel;
+        // Damp normal velocity according to parameter
+        nVel *= restitutionCoefficientN;
+        // Combine
+        velocity[planeConstraints.particles[i]] = tVel + nVel;
```

Comment:
I was more referencing the way the vector is broken up and 
`dot(nVel, planeConstraints.normals[i]) > 0 ? nVel : -nVel;`